### PR TITLE
fix(react-radio): Allow data argument on onChange

### LINKED ISSUE #22752: [Bug]: react-radio typings for onChange does not allow data argument
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
-
```

### Are you reporting Accessibility issue?

no

### Reproduction

-

## Actual Behavior
`RadioProps` types `onChange` as it has two arguments:
```ts
  onChange?: (ev: React.ChangeEvent<HTMLInputElement>, data: RadioOnChangeData) => void;
```

As the `RadioProps` is intersection of `ComponentProps<..., 'input'>` it does not allow the `data` argument:
```

TS2322: Type '(e: ChangeEvent<HTMLInputElement>, d: RadioOnChangeData) => void' is not assignable to type 'ChangeEventHandler<HTMLInputElement> & ((ev: ChangeEvent<HTMLInputElement>, data: RadioOnChangeData) => void)'.   Type '(e: ChangeEvent<HTMLInputElement>, d: RadioOnChangeData) => void' is not assignable to type 'ChangeEventHandler<HTMLInputElement>'.  index.d.ts(2217, 9): The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & Omit<ComponentProps<Partial<RadioSlots>, "input">, "size"> & { value?: string | undefined; labelPosition?: "after" | ... 1 more ... | undefined; disabled?: boolean | undefined; onChange?: ((ev: ChangeEvent<...>, data: RadioOnChangeData) => void) | undefined; } & RefAttributes<...>'
```

## Expected Behavior
`data` argument is valid on `onChange`

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by miroslavstastny**
## Current Behavior

`Radio.onChange(event, data)` results in typescript error, `data` argument is not allowed:

```
TS2322: Type '(e: ChangeEvent<HTMLInputElement>, d: RadioOnChangeData) => void' is not assignable to type 'ChangeEventHandler<HTMLInputElement> & ((ev: ChangeEvent<HTMLInputElement>, data: RadioOnChangeData) => void)'.   Type '(e: ChangeEvent<HTMLInputElement>, d: RadioOnChangeData) => void' is not assignable to type 'ChangeEventHandler<HTMLInputElement>'.  index.d.ts(2217, 9): The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & Omit<ComponentProps<Partial<RadioSlots>, "input">, "size"> & { value?: string | undefined; labelPosition?: "after" | ... 1 more ... | undefined; disabled?: boolean | undefined; onChange?: ((ev: ChangeEvent<...>, data: RadioOnChangeData) => void) | undefined; } & RefAttributes<...>'
```

## New Behavior

`Radio.onChange` now has two arguments - `event` and `data`.

## Related Issue(s)

Fixes #22752

### CODE REVIEW

File: packages/react-radio/src/components/Radio/Radio.types.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export type RadioSlots = {
 /**
  * Radio Props
  */
-export type RadioProps = Omit<ComponentProps<Partial<RadioSlots>, 'input'>, 'size'> & {
+export type RadioProps = Omit<ComponentProps<Partial<RadioSlots>, 'input'>, 'onChange' | 'size'> & {
```
Comment (miroslavstastny) [inline packages/react-radio/src/components/Radio/Radio.types.ts:36]: This approach is supposed to be consistent with `InputProps` in `react-input`.

**Review by GeoffCoxMSFT — APPROVED** [review]

**Comment by GeoffCoxMSFT** [comment]
@micahgodbolt - you are helping out with radio and checkbox right? If so, you should review too :)

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                 |        Size (minified/GZIP) |
| ------------------------------------------------------------------------------------------------- | --------------------------: |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>           | `23.556 kB`<br />`7.898 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr> |  `8.205 kB`<br />`3.516 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/7ad290b51135cb84c84e7dd8771bd712892c7484'>7ad290b51135cb84c84e7dd8771bd712892c7484</a></sub>
<div id="bundleSizeComment9423"/>

**Review by spmonahan — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/Avatar-mount_base.html">1130</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/Avatar-mount.html">1101</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/Button-mount_base.html">662</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/Button-mount.html">668</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProvider-mount_base.html">2206</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProvider-mount.html">2210</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProviderWithTheme-mount_base.html">334</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProviderWithTheme-mount.html">350</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">295</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">281</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">391</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">360</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/MakeStyles-mount_base.html">1861</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22753/perf-test-react-components/MakeStyles-mount.html">1839</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by khmakoto — APPROVED** [review]

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**